python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py

```diff
@@ -340,3 +343,108 @@ def service_client(self) -> DataServiceClient:
+        self._default_required_access_type = None
+
+        self._scheduler_client = None
+        """SchedulerClient: Client for interacting with scheduler, which also is a context manager for connections."""
+
+    def _generate_request_response(self, request: Union[JobControlRequest, JobInfoRequest, JobListRequest],
+                                   success: bool, reason: str, message: Optional[str] = None) -> Union[JobControlResponse, JobInfoResponse, JobListResponse]:
+        """
+        Generate an appropriate response object for the supplied request.
+
+        Parameters
+        ----------
+        request: Union[JobControlRequest, JobInfoRequest, JobListRequest]
+            A request instance of one of the valid types for this instance.
+        success: bool
+            Whether the response should indicate success.
+        reason: str
+            The summary reason for success or failure in the response.
+        message: Optional[str]
+            An optional, more detailed message on success or failure for the response.
+
+        Returns
+        -------
+        Union[JobControlResponse, JobInfoResponse, JobListResponse]
+            An appropriate response object.
+        """
+        if isinstance(request, JobControlRequest):
+            return JobControlResponse(action=request.action, job_id=request.job_id, success=success, reason=reason,
+                                      message=message)
+        elif isinstance(request, JobInfoRequest):
+            return JobInfoResponse(job_id=request.job_id, status_only=request.status_only, success=success,
+                                   reason=reason, message=message)
+        elif isinstance(request, JobListRequest):
+            return JobListResponse(only_active=request.only_active, success=success, reason=reason, message=message)
+        else:
+            raise TypeError(f"Invalid message type {request.__class__.__name__} sent to {self.__class__.__name__}")
+
+    async def determine_required_access_types(self, request: ExternalRequest, user) -> tuple:
+        """
+        Determine what access is required for this request from this user to be accepted.
+
+        Determine the necessary access types for which the given user needs to be authorized in order for the user to
+        be allow to submit this request, in the context of the current state of the system.
+
+        Parameters
+        ----------
+        request
+        user
+
+        Returns
+        -------
+        A tuple of required access types required for authorization for the given request at this time.
+        """
+        # TODO: implement something
+        # TODO: may have to start to track both access level and job "ownership"
+        # FIXME: for now, just use the default type (which happens to be "everything")
+        return self._default_required_access_type,
+
+        if not isinstance(request, (JobControlRequest, JobInfoRequest, JobListRequest)):
```

Comment:
This is most likely a merge issue.

Reply:
I think so too.  Just cleaned it up.  Hopefully the tests will pass in a few minutes and this will be good to go.